crates/bevy_window/src/window.rs

```diff
@@ -90,6 +93,34 @@ impl WindowResizeConstraints {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct WindowIconBytes {
+    pub bytes: Vec<u8>,
+    pub width: u32,
+    pub height: u32,
+}
+
+#[derive(Debug, Clone)]
+pub enum WindowIcon {
+    Path(PathBuf),
+    Bytes(WindowIconBytes),
+}
+
+impl<T> From<T> for WindowIcon
+where
+    T: AsRef<Path>,
```

Comment:
```suggestion
    T: Into<PathBuf>,
```

We should avoid implicit allocations here if possible.

Reply:
For example, someone could call:
```rust
let pathbuf = PathBuf::from("hello.txt");
let icon = WindowIcon::from(&pathbuf);
```

Since `PathBuf` implements `AsRef<Path>`, this implementation would duplicate the allocated memory from the original `PathBuf` even though we don't need it anymore.

Reply:
Thanks for the explanatory comment. 
Makes sense to me now.